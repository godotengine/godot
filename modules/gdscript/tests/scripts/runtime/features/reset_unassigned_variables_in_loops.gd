# GH-56223, GH-76569

func test():
	for i in 3:
		var a
		if true:
			var b
			if true:
				var c
				@warning_ignore("unassigned_variable")
				prints("Begin:", i, a, b, c)
				a = 1
				b = 1
				c = 1
				prints("End:", i, a, b, c)
	print("===")
	var j := 0
	while j < 3:
		var a
		if true:
			var b
			if true:
				var c
				@warning_ignore("unassigned_variable")
				prints("Begin:", j, a, b, c)
				a = 1
				b = 1
				c = 1
				prints("End:", j, a, b, c)
		j += 1
