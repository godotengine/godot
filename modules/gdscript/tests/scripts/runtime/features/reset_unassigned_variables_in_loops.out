GDTEST_OK
Begin: 0 <null> <null> <null>
End: 0 1 1 1
Begin: 1 <null> <null> <null>
End: 1 1 1 1
Begin: 2 <null> <null> <null>
End: 2 1 1 1
===
Begin: 0 <null> <null> <null>
End: 0 1 1 1
Begin: 1 <null> <null> <null>
End: 1 1 1 1
Begin: 2 <null> <null> <null>
End: 2 1 1 1
